statviz/src/utils/theme.ts

```diff
@@ -213,3 +214,230 @@ export const nivoScheme: Theme = {
   },
 };
+
+export const scaledNivoTheme = (width: number, height: number, scaleFactor: number = 5): Theme => {
+  const strokeWidth = Math.ceil(height / 900);
+  const fontSizeAxis = Math.ceil(height / 35);
+  const fontSizeLegend = Math.ceil(height / 35);
+  const fontSizeText = Math.ceil(height / 35);
+  const fontSizeLabel = Math.ceil(width / (scaleFactor * 4.5));
+
+  return {
+    background: "#ffffff",
+    fontFamily: "Open Sans",
+    text: {
+      fontSize: fontSizeText,
+      fill: "#333333",
+      outlineWidth: 0,
+      outlineColor: "transparent",
+    },
+    labels: {
+      text: {
+        fontSize: fontSizeLabel,
+      },
+    },
+    axis: {
+      domain: {
+        line: {
+          stroke: "#777777",
+          strokeWidth,
+        },
+      },
+      legend: {
+        text: {
+          fontSize: fontSizeLegend,
+          fill: "#333333",
+          outlineWidth: 0,
+          outlineColor: "transparent",
+        },
+      },
+      ticks: {
+        line: {
+          stroke: "#777777",
+          strokeWidth,
+        },
+        text: {
+          fontSize: fontSizeAxis,
+          fill: "#333333",
+          outlineWidth: 0,
+          outlineColor: "transparent",
+        },
+      },
+    },
+    grid: {
+      line: {
+        stroke: "#dddddd",
+        strokeWidth,
+      },
+    },
+    legends: {
+      title: {
+        text: {
+          fontSize: fontSizeLegend,
+          fill: "#333333",
+          outlineWidth: 0,
+          outlineColor: "transparent",
+        },
+      },
+      text: {
+        fontSize: fontSizeLegend,
+        fill: "#333333",
+        outlineWidth: 0,
+        outlineColor: "transparent",
+      },
+      ticks: {
+        line: {},
+        text: {
+          fontSize: fontSizeLegend,
+          fill: "#333333",
+          outlineWidth: 0,
+          outlineColor: "transparent",
+        },
+      },
+    },
+    annotations: {
+      text: {
+        fontSize: fontSizeLegend,
+        fill: "#333333",
+        outlineWidth: 2,
+        outlineColor: "#ffffff",
+        outlineOpacity: 1,
+      },
+      link: {
+        stroke: "#000000",
+        strokeWidth: 1,
+        outlineWidth: 2,
+        outlineColor: "#ffffff",
+        outlineOpacity: 1,
+      },
+      outline: {
+        stroke: "#000000",
+        strokeWidth: 2,
+        outlineWidth: 2,
+        outlineColor: "#ffffff",
+        outlineOpacity: 1,
+      },
+      symbol: {
+        fill: "#000000",
+        outlineWidth: 2,
+        outlineColor: "#ffffff",
+        outlineOpacity: 1,
+      },
+    },
+    tooltip: {
+      container: {
+        background: "#ffffff",
+        fontSize: fontSizeLegend,
+      },
+      basic: {},
+      chip: {},
+      table: {},
+      tableCell: {},
+      tableCellValue: {},
+    },
+  };
+};
+
+// Calculate margins, fontSizes, and styles for additional information that can be displayed in an exported image
+export const graphMarginTopPercent = 10;
```

Comment:
I would split this up in nivo theme, visx theme, chakra theme and maybe styling helpers for exporting